js-weather/assets/javascript/weather_fetcher.js

```diff
@@ -0,0 +1,52 @@
+export class WetherFetcher {
+  constructor() {
+    this.appId = "4b5774e9f3d2a07b84f0f2f88e486224";
+  }
+
+  async fetchByXMLHttpRequest(city) {
+    const requestUrl = this._createUrl(city);
+    return new Promise((resolve, reject) => {
+      const xhr = new XMLHttpRequest();
+      xhr.open("GET", requestUrl);
+      xhr.send();
+      xhr.onreadystatechange = () => {
+        if (xhr.readyState === 4 && xhr.status === 200) {
+          const json = JSON.parse(xhr.responseText);
+          resolve(json);
+        } else if (xhr.status !== 200) {
+          reject();
+        }
+      };
+    })
+      .then((json) => {
+        console.log("XMLHttpRequestオブジェクトの通信は成功したよ");
+        return json;
+      })
+      .catch(() => {
+        console.log("XMLHttpRequestオブジェクトの通信は失敗したよ");
+      });
+  }
+
+  async fetchByAsyncAwait(city) {
+    const requestUrl = this._createUrl(city);
+    return await fetch(requestUrl)
```

Comment:
(修正不要)
ここのasyncとawaitは不要ですね。

```suggestion
  fetchByAsyncAwait(city) {
    const requestUrl = this._createUrl(city);
    return fetch(requestUrl)
```